### PR TITLE
Implement drivers command

### PR DESCRIPTION
**Pull request by ssbarnea**
New command lists available drivers by running sanity checks on
them. Only drivers that are configured are supposed to be returned.

Sample output:
```
$ molecule drivers                                                                                                                                                                           --> Sanity checks: 'docker'
--> Sanity checks: 'hetznercloud'
Driver hetznercloud not loaded due: 1
--> Sanity checks: 'podman'
ERROR: Missing Hetzner Cloud driver dependency. Please install via 'molecule[hetznercloud]' or refer to your INSTALL.rst driver documentation file
name
------------
azure
delegated
digitalocean
docker
ec2
gce
linode
lxc
lxd
openstack
podman
vagrant
```

Part of #2228

### CODE REVIEW

**Comment by ssbarnea** [comment]
@cloudnull @ekultails @gabrielecerami please review this. thanks.

File: molecule/config.py [hunk 1]
```diff
@@ -484,22 +491,36 @@ def molecule_file(path):
...
-        gce.GCE(None).name,
-        hetznercloud.HetznerCloud(None).name,
-        linode.Linode(None).name,
-        lxc.LXC(None).name,
-        lxd.LXD(None).name,
-        openstack.Openstack(None).name,
-        podman.Podman(None).name,
-        vagrant.Vagrant(None).name,
+def molecule_drivers(validate=False):
+    # TODO(ssbarnea)
```
Comment (fabianvf) [inline molecule/config.py]: What is this Todo for?

File: molecule/config.py [hunk 2]
```diff
@@ -484,22 +491,36 @@ def molecule_file(path):
...
     ]
+    drivers = []
+    cfg = Config(None)
+    for x in driver_names:
+        d = getattr(globals()[x.lower()], x)(cfg)
+        try:
+            if validate and d.sanity_checks(force=True) is not None:
+                continue
+        except (Exception, SystemExit) as e:
+            LOG.warn("Driver {0} not loaded due: {1}".format(d.name, e))
```
Comment (fabianvf) [inline molecule/config.py]: `s/due/due to/`

File: molecule/command/drivers.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    default='simple',
+    help='Change output format. (simple)',
+)
+def drivers(ctx, format):  # pragma: no cover
+    """ Lists drivers. """
+
+    drivers = [[x] for x in config.molecule_drivers(validate=True)]
+
+    headers = ['name']
+    if format == 'simple' or format == 'plain':
```
Comment (fabianvf) [inline molecule/command/drivers.py]: Is this conditional required? I think format has to be one of these anyway

Comment (ssbarnea) [reply molecule/command/drivers.py]: Right, a copy/paste leftover from the implementation time, fixing it.

**Pull request merged**